Source/Core/Core/Boot/Boot.cpp

```diff
@@ -38,6 +42,76 @@
 #include "DiscIO/NANDContentLoader.h"
 #include "DiscIO/Volume.h"
 
+BootParameters::BootParameters(Parameters&& parameters_) : parameters(std::move(parameters_))
+{
+}
+
+std::unique_ptr<BootParameters> BootParameters::GenerateFromFile(const std::string& path)
+{
+  const bool is_drive = cdio_is_cdrom(path);
+  // Check if the file exist, we may have gotten it from a --elf command line
+  // that gave an incorrect file name
+  if (!is_drive && !File::Exists(path))
+  {
+    PanicAlertT("The specified file \"%s\" does not exist", path.c_str());
+    return {};
+  }
+
+  std::string extension;
+  SplitPath(path, nullptr, nullptr, &extension);
+  std::transform(extension.begin(), extension.end(), extension.begin(), ::tolower);
+
+  const std::unordered_set<std::string> disc_image_extensions = {
```

Comment:
Whoops, looks like I forgot to add the static when I moved it into the function. Fixed.